core/connection.js

```diff
@@ -619,3 +621,32 @@ Blockly.Connection.prototype.setShadowDom = function(shadow) {
 Blockly.Connection.prototype.getShadowDom = function() {
   return this.shadowDom_;
 };
+
+/**
+ * @return {string} A human-readable description of this Connection.
+ */
+Blockly.Connection.prototype.describe = function() {
```

Comment:
As noted, Block.toString() is already in use. Given the shared intentions between the methods, it seems best to use a different name.